src/universe/UniverseTypeValidator.java

```diff
@@ -0,0 +1,100 @@
+package universe;
+
+import com.sun.source.tree.ParameterizedTypeTree;
+import com.sun.source.tree.Tree;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.common.basetype.BaseTypeValidator;
+import org.checkerframework.common.basetype.BaseTypeVisitor;
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.AnnotatedTypeParameterBounds;
+import org.checkerframework.framework.util.AnnotatedTypes;
+import org.checkerframework.javacutil.TreeUtils;
+
+import javax.lang.model.element.TypeElement;
+import java.util.List;
+
+import static universe.UniverseInferenceChecker.BOTTOM;
+import static universe.UniverseInferenceChecker.LOST;
+import static universe.UniverseInferenceChecker.REP;
+
+/**
+ * This type validator ensures correct usage of ownership modifiers or generates
+ * constraints to ensure well-formedness.
```

Comment:
The 'generates constraints` part here is odd, as this is the normal type validator, not for inference.